case_studies/galaxy_clustering/encoder/encoder.py

```diff
@@ -19,36 +18,21 @@ def initialize_networks(self):
         power_of_two = (self.tile_slen != 0) & (self.tile_slen & (self.tile_slen - 1) == 0)
         assert power_of_two, "tile_slen must be a power of two"
         ch_per_band = sum(inorm.num_channels_per_band() for inorm in self.image_normalizers)
-        num_features = 256
-        self.features_net = GalaxyClusterFeaturesNet(
+        self.net = GalaxyClusterFeaturesNet(
             len(self.survey_bands),
             ch_per_band,
-            num_features,
+            self.var_dist.n_params_per_source,  # output dimension
             tile_slen=self.tile_slen,
             downsample_at_front=self.downsample_at_front,
         )
-        n_params_per_source = self.var_dist.n_params_per_source
-        self.catalog_net = CatalogNet(num_features, n_params_per_source)
```

Comment:
@kapnadak @gapatron `CatalogNet` is no longer around. To account for this, I adjusted your code by removing its references to `CatalogNet` and by making the output of your `GalaxyClusterFeaturesNet` have the right output dimension to be used directly to parameterize the variational distribution. I haven't tested these changes though: some additional modifications may be needed.